### ISSUE TITLE
NRK logos undermerkene / radiokanalene

### ISSUE THREAD

**Issue by mengeba** [nrkno/core-icons#198]
**Do you have an svg icon file?**
Please upload this file as an attachment or name the upstream branch containing it.

**Is the icon file ready to be added to the library?**
 - It has been trimmed using a minifier like [SVGOMG](https://jakearchibald.github.io/svgomg/). (Lowest precision without visual degradation)
- It has fill-rule of `fill="currentColor"`
- It has no explicit width/height
- It has viewbox of `viewbox="0 0 24 24"`

**What is the name of the icon?**
Make sure the icon fits the naming convention of `nrk-category-name--state`

**Additional context**
Lagt til branch med nye logoer til undermerkene / radiokanalene. default+optisk justert. beskrivelse for minimumstørrelse for bruk ligger her https://portalen.nrk.no/design/merking/

**Comment by asmundmatzow** [comment]
@marve-svenningsen has defined new logos in Figma: https://www.figma.com/file/HYTp671vlyywihrW465Hpb/Undermerker?node-id=769%3A10183
![Undermerker](https://user-images.githubusercontent.com/7030680/200785186-4c45bfe1-5e1b-4048-8fa7-d55c49256398.png)

Variables:
* Size: small, large
* Layout: default, stacked
* Color: default, light, dark

**Comment by asmundmatzow** [comment]
Some thoughts on variables after discussing with Designutvikling team:

- The variables and options should make semantic sense, and have a consistent internal logic.
- All variable names and options should use lowercase.
- Ideally the "size" parameter should convey that this is about optimzing logo rendering for use in small sizes vs. large sizes, but "size:small/large" might be ok, as long as we can link to documentation. Alternative might be "render size:small/large"
- the "layout" variable can probably contain the "slab" parameter, although this also changes color – the fact that it changes the bounding box takes precedence.
- The default "layout" option should not be named "default", but something more descriptive. The "stacked" option should maybe be called something else (it is possible to stack things both horizontally and vertically). Possibilities might be to call "default" "horizontal" and "stacked" "vertical", or "landscape" and "portrait", or "laying" and "standing", or "wide" and "narrow".
- For "color" it is ok to keep the "default" option (this will not be called in code – no option=default). But the variable should be renamed to "theme", and the options renamed from "light" to "darkmode" and from "dark" to "lightmode".

**Comment by asmundmatzow** [comment]
Also, the "slab" option could potentially get a better name, the describes it's use more clearly. Maybe "bg-box", "boxed-wide" or something along those lines. Ideas are appreciated.

In general it might be worth discussing if variables/options should describe concrete visual appearance, or rather intended use. As in "logo with slab" vs "use where one does not have control of the background color". An obvious problem with naming things for use is to make it short enough to be workable as a variable option, and still convey a clear meaning.

**Comment by skjalgepalg** [comment]
Using NRK Klassisk as an example to illustrate _possible_ names/use in the various code contexts. Input/feedback is most welcome!

# General naming structure

* icons: `nrk-category-name--state`
* logos: `(nrk-logo-)nrk-brand-rendersize-layout--color` -Uncertain if we need the `nrk-logo` prefix
  * If we see the need, we could append specific orientation to slab/boxed variants as they have no color variations:
  * boxed-logos: `(nrk-logo-)nrk-brand-rendersize-boxed--orientation`

## HTML

In HTML this translates quite directly to the id-structure, keeping the kebab-case:

* `<use xlink:href="#nrk-logo-nrk-klassisk-small-horizontal"></use>`
* `<use xlink:href="#nrk-logo-nrk-klassisk-small-horizontal--darkmode"></use>`
* `<use xlink:href="#nrk-logo-nrk-klassisk-small-horizontal--lightmode"></use>`
* `<use xlink:href="#nrk-logo-nrk-klassisk-small-boxed"></use>`

## JavaScript
We use camelCase as imports are not overly fond of kebab-case

```JavaScript
import {
  nrkKlassiskSmallVertical, // Stacked in plain white
  nrkKlassiskSmallVerticalDarkmode, // Stacked for use in dark-theme
  nrkKlassiskSmallVerticalLightmode, // Stacked for use in light-theme
  nrkKlassiskSmallHorizontal, // Horizontal in plain white
  nrkKlassiskSmallHorizontalDarkmode, // Horizontal for use in dark-theme
  nrkKlassiskSmallHorizontalLightmode, // Horizontal for use in light-theme
  nrkKlassiskSmallBoxedWide, // Slab in horizontal/wide
  // nrkKlassiskSmallBoxedSplit, // Slab with split color
} from 'core-icons/logo'
```

## JavaScript (React/JSX)

Basically the same but with different source folder and using PascalCase

```JSX
import {
  NrkKlassiskSmallVertical, // Stacked in plain white
  NrkKlassiskSmallVerticalDarkmode, // Stacked for use in dark-theme
  NrkKlassiskSmallVerticalLightmode, // Stacked for use in light-theme
  NrkKlassiskSmallHorizontal, // Horizontal in plain white
  NrkKlassiskSmallHorizontalDarkmode, // Horizontal for use in dark-theme
  NrkKlassiskSmallHorizontalLightmode, // Horizontal for use in light-theme
  NrkKlassiskSmallBoxedWide, // Slab in horizontal/wide
  // NrkKlassiskSmallBoxedSplit, // Slab in plain white
} from 'core-icons/logo/jsx'
```

**Comment by asmundmatzow** [comment]
After some more discussion in the Designutvikling team, we have landed on this naming structure, which only changes the naming, not where variables are placed:

1. rendersize: s, l
2. layout: title-right, title-below, with-bg
3. mode: default, on-light, on-dark

<img width="1768" alt="Screenshot 2022-11-11 at 15 01 15" src="https://user-images.githubusercontent.com/7030680/201356480-967ce337-696f-4fbe-8a14-6b14608c55c2.png">

**Comment by asmundmatzow** [comment]
Decided to create a separate folder (tentatively named "preview") next to "logo" and "expressive", for logos and icons that are available for testing, but not cleared for production.

**Comment by asmundmatzow** [comment]
These logos are published in [Core Icons v13](https://github.com/nrkno/core-icons/pull/223).
